### PR TITLE
Enables more wasm backend tests

### PR DESCRIPTION
**Pull request by jgravelle-google**
Most of these were fixed via various changes over the past month or so.

### CODE REVIEW

File: cmake/Modules/Platform/Emscripten.cmake [hunk 1]
```diff
@@ -134,8 +134,6 @@ if (EMSCRIPTEN_FORCE_COMPILERS)
 	set(CMAKE_CXX_COMPILER_ID Clang)
 	set(CMAKE_CXX_STANDARD_COMPUTED_DEFAULT 98)
 
-	set(CMAKE_C_COMPILER_TARGET "asmjs-unknown-emscripten")
```
Comment (dschuff) [inline cmake/Modules/Platform/Emscripten.cmake]: is this actually necessary? In particular, have you checked that CMake correctly infers the target when it queries the compiler? I wouldn't be at all surprised if its check doesn't work right due to the quirks of emscripten.

Comment (kripken) [inline cmake/Modules/Platform/Emscripten.cmake]: @juj, do we need these?

**Comment by kripken** [comment]
Looks ok aside from that question for @juj. I assume it's ok to remove those, but maybe there's some cmake issue?

File: cmake/Modules/Platform/Emscripten.cmake (hunk 1 above)
Comment (jgravelle-google) [inline cmake/Modules/Platform/Emscripten.cmake]: Before we'd call `wasm-install/bin/clang -target wasm32-unknown-unknown [...] --triple asmjs-unknown-emscripten`
Now we call `wasm-install/bin/clang -target wasm32-unknown-unknown`, no `--triple`
And then watching the output for asm2wasm it calls `-target asmjs-unknown-emscripten`, and builds correctly.

There might be some case I'm not considering though.

Comment (dschuff) [inline cmake/Modules/Platform/Emscripten.cmake]: If the only difference is the invocation of clang, then `--triple` is unnecessary even in the asmjs case. But I'm not sure if `CMAKE_C_COMPILER_TARGET` also has some other effect on CMake that I don't know about.

**Comment by dschuff** [comment]
Can we add the `__unix__` thing as a separate PR? (is that what makes a lot of these tests pass btw?)

**Comment by jgravelle-google** [comment]
It's the second piece of what makes the zlib test actually pass. That is also affected by the Emscripten.cmake changes I believe? I can make it a separate PR.

More context for everyone else: https://reviews.llvm.org/D26431
tl;dr, Emscripten provides a unixy environment, so it makes sense to pretend that wasm is unixy, but only in Emscripten itself.

**Comment by jgravelle-google** [comment]
Split the `__unix__` bit out into https://github.com/kripken/emscripten/pull/4718

File: cmake/Modules/Platform/Emscripten.cmake (hunk 1 above)
Comment (jgravelle-google) [inline cmake/Modules/Platform/Emscripten.cmake]: Looking at the documentation: https://cmake.org/cmake/help/v3.0/manual/cmake-toolchains.7.html
It sounds like it's just used for specifying the triple, which emscripten already explicitly handles.

Comment (dschuff) [inline cmake/Modules/Platform/Emscripten.cmake]: Hm, OK. In the fullness of time I actually do think it makes sense to have this in the CMake description file because that would allow use of a system-installed clang that supports native or wasm. But that would be after triple convergence, and maybe would be  different CMake file than this one anyway, so I'm OK dropping this now, if @kripken and/or @juj  are.

**Comment by kripken** [comment]
Overall this looks ok to me, except for the cmake parts that I don't understand and @juj should take a look at. Note that @juj is traveling, so if this is urgent for you, feel free to land it for now.

File: cmake/Modules/Platform/Emscripten.cmake (hunk 1 above)
Comment (juj) [inline cmake/Modules/Platform/Emscripten.cmake]: @jgravelle-google : This is ok to remove, as long as tests `other.test_cmake*` pass after this change. Was this verified?

The only visible difference to this change is I believe that these fields will come out empty to user CMakeLists.txt files, so user projects will see these being absent for them. This should not be a big issue though, the origin of these variables in CMake was to be used by toolchain file implementors, and not by CMake project build script writers.

**Comment by juj** [comment]
Can you squash the changes, since there's a bit back and forth with reverts? Otherwise lgtm, as long as the tests `other.test_cmake*` still pass.

**Comment by jgravelle-google** [comment]
Tests pass, squashed+rebased
